napari/layers/graph/graph.py

```diff
@@ -435,6 +468,15 @@ def _remove_nodes(
                 )
             )
 
+        self.events.data(
+            value=self.data,
+            action=ActionType.REMOVING,
+            data_indices=tuple(
+                indices,
```

Comment:
`indices` could be either in the buffer or world domain depending on `is_buffer_domain` value, so the emitted value will not be consistent.

I suggest creating `world_indices = self.data._buffer2world[indices] if is_buffer_domain else indices` and emitting that.